Source/WebCore/rendering/svg/SVGRenderSupport.cpp

```diff
@@ -141,25 +141,33 @@ LayoutRepainter::CheckForRepaint SVGRenderSupport::checkForSVGRepaintDuringLayou
     return LayoutRepainter::CheckForRepaint::Yes;
 }
 
-auto SVGRenderSupport::computeContainerBoundingBoxes(const RenderElement& container, RepaintRectCalculation repaintRectCalculation) -> ContainerBoundingBoxes
+static bool hasValidBoundingBoxForContainer(const RenderObject& object)
 {
-    ContainerBoundingBoxes result;
+    if (auto* svgShape = dynamicDowncast<LegacyRenderSVGShape>(object))
+        return svgShape && !svgShape->isRenderingDisabled();
 
-    for (CheckedRef current : childrenOfType<RenderObject>(container)) {
-        if (current->isLegacyRenderSVGHiddenContainer())
-            continue;
+    if (auto* text = dynamicDowncast<RenderSVGText>(object))
+        return text && text->isObjectBoundingBoxValid();
 
-        // Don't include elements in the union that do not render.
-        if (auto* shape = dynamicDowncast<LegacyRenderSVGShape>(current.ptr()); shape && shape->isRenderingDisabled())
-            continue;
+    if (auto* hiddenContainer = dynamicDowncast<LegacyRenderSVGContainer>(object))
+        return hiddenContainer && !hiddenContainer->isLegacyRenderSVGHiddenContainer();
 
-        if (auto* text = dynamicDowncast<RenderSVGText>(current.ptr()); (text && !text->isObjectBoundingBoxValid()))
-            continue;
+    if (auto* foreignObject = dynamicDowncast<LegacyRenderSVGForeignObject>(object))
+        return foreignObject && foreignObject->isObjectBoundingBoxValid();
 
-        if (auto* image = dynamicDowncast<LegacyRenderSVGImage>(current.ptr()); (image && !image->isObjectBoundingBoxValid()))
-            continue;
+    if (auto* svgImage = dynamicDowncast<LegacyRenderSVGImage>(object))
+        return svgImage && svgImage->isObjectBoundingBoxValid();
 
-        if (auto* foreignObject = dynamicDowncast<LegacyRenderSVGForeignObject>(current.ptr()); (foreignObject && !foreignObject->isObjectBoundingBoxValid()))
+    return false;
+}
+
+auto SVGRenderSupport::computeContainerBoundingBoxes(const RenderElement& container, RepaintRectCalculation repaintRectCalculation) -> ContainerBoundingBoxes
+{
+    ContainerBoundingBoxes result;
+
+    for (CheckedRef current : childrenOfType<RenderObject>(container)) {
+        // Don't include elements that are not rendered in the union.
+        if (!hasValidBoundingBoxForContainer(current))
```

Comment:
It’s a bad sign that the comment says “elements that are not rendered” but the function name says “has valid bounding box”. There’s no reason to not name the code the same way we’d describe it to a person. Then we wouldn’t need a comment because the code would say the same thing as the comment! So either the comment should change to match the function name and then perhaps be removed, or the function name should change to match the comment and then perhaps the comment would be removed.

If we keep the comment, it should explain _why_ the code does what it does, not just repeat what the code does with different words.

Reply:
I removed this comment but added link to specification which describes what we are doing in `hasValidBoundingBoxForContainer` function to have valid bounding box for container. It can make it easier to understand this code.